version: "3"

networks:
  gitea:
    external: false

volumes:
  gitea-app:
    driver: local
  gitea-db:
    driver: local

services:
  app:
    image: gitea/gitea:1
    container_name: gitea-app
    environment:
      - USER_UID=1000
      - USER_GID=1000
      - GITEA__database__DB_TYPE=postgres
      - GITEA__database__HOST=db:5432
      - GITEA__database__NAME=gitea
      - GITEA__database__USER=gitea
      - GITEA__database__PASSWD=gitea
    restart: unless-stopped
    networks:
      - gitea
    volumes:
      - gitea-app:/data
      - /etc/timezone:/etc/timezone:ro
      - /etc/localtime:/etc/localtime:ro
    ports:
      - "3000:3000"
      - "222:22"
    healthcheck:
      test: [CMD, "curl", "-f", "http://127.0.0.1:3000" ]
      timeout: 1s
      interval: 5s
      retries: 10
    depends_on:
      - db

  db:
    image: postgres:16
    container_name: gitea-db
    restart: unless-stopped
    environment:
      - POSTGRES_USER=gitea
      - POSTGRES_PASSWORD=gitea
      - POSTGRES_DB=gitea
    networks:
      - gitea
    volumes:
      - gitea-db:/var/lib/postgresql/data
